src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -484,19 +485,28 @@ void Unit::UpdateSplineMovement(uint32 t_diff)
     movespline->updateState(t_diff);
     bool arrived = movespline->Finalized();
 
+    if (movespline->isCyclic())
+    {
+        m_splineSyncTimer.Update(t_diff);
+        if (m_splineSyncTimer.Passed())
+        {
+            m_splineSyncTimer.Reset(5000); // Retail value, do not change
+
+            WorldPacket data(SMSG_FLIGHT_SPLINE_SYNC, 4 + GetPackGUID().size());
+            Movement::PacketBuilder::WriteSplineSync(*movespline, data);
+            data.appendPackGUID(GetGUID());
+            SendMessageToSet(&data, true);
+        }
+    }
+
     if (arrived)
         DisableSpline();
 
-    m_movesplineTimer.Update(t_diff);
-    if (m_movesplineTimer.Passed() || arrived)
```

Comment:
could this timer be restored to avoid calling UpdateSplinePosition() every update ?

Reply:
It's less about updating the position, it's about the following triggered actions such as vmap based position data updates. Long story short, this change is not only updating spline positions but also causing vmap lookups which are now being spamed with every single update tick. Not really a healthy thing to do

Reply:
Then maybe that's the thing you want to optimize instead? Throttle (or completely block) movement flags recalc for creatures who are using random or waypoint movement out of combat, throttle visibility updates and MoveInLineOfSight calls. IMO that's a better solution than causing server/client desyncs.

And as for dynamic spawns, I don't know how they work, but I cannot see why can zone be changeable in real time: blizzard used separate servers for separate zones from the beginning, as they themselves said, and I highly doubt that it makes sense for spawns to be able to transition from zone to zone under these circumstances, as that would mean they would have to be moved to a different server responsible for that zone, which is just a technical overkill.

Reply:
this can be related with this issue?
https://github.com/TrinityCore/TrinityCore/issues/24356

Reply:
If the claim "seems related with pathfinding" is true - then no. Pathfinding would be executed in the same amounts with or without this position update delay. This delay is only responsible for controlling how often is the unit being physically moved along the already generated path.